### PR TITLE
Implement #elifdef and #elifndef

### PR DESCRIPTION
**Pull request by danielsan901998**
Implement C23 feature #elifdef and #elifndef from #430.
For testing added two new Preproccesor tests [#elifdef.c](https://github.com/Vexu/arocc/compare/master...danielsan901998:arocc:master?expand=1#diff-d863a9d1369bd4db5960cf805446a9c5a247128c4780dfc6088d0d595d2e5ab2) and [#elifdefc2x.c](https://github.com/Vexu/arocc/compare/master...danielsan901998:arocc:master?expand=1#diff-b705cf5e912d8d27e52ba76e1113f7d680dd3795c2ecf374d633794483cecb29) to check it is proccessed correctly with the -std=c2x flag
Also changed build.zig to compile with current master branch of zig.

### CODE REVIEW

File: src/Preprocessor.zig [hunk 1]
```diff
@@ -365,6 +365,72 @@ fn preprocessExtra(pp: *Preprocessor, source: Source) MacroError!Token {
...
+                        if (if_level == 0) {
+                            try pp.err(directive, .elifdef_without_if);
+                            if_level += 1;
+                            if_kind.set(if_level, until_else);
+                        } else if (if_level == 1) {
+                            guard_name = null;
+                        }
+                        switch (if_kind.get(if_level)) {
+                            until_else => {
+                                const macro_name = (try pp.expectMacroName(&tokenizer)) orelse continue;
```
Comment (Vexu) [inline src/Preprocessor.zig:378]: I think a missing macro name should be considered `false` similarly to a missing expression with `#if`.

**Review by Vexu — APPROVED** [review]
Looks good! And thanks for updating this for the latest master.

File: src/Preprocessor.zig (hunk 1 above)
Comment (danielsan901998) [reply src/Preprocessor.zig]: I have tested it with clang and you are right, i have also added a test for checking the missing macro name in the commit ef844b664b3ff4f6a025ae415669c94f2665ac22

Comment (danielsan901998) [reply src/Preprocessor.zig]: Clang generate this output:
```cpp
test/cases/#elifdefc2x_error.c:7:9: error: macro name missing
#elifdef
        ^
# 10 "test/cases/#elifdefc2x_error.c"
inttest/cases/#elifdefc2x_error.c:16:10: error: macro name missing
#elifndef
         ^

# 19 "test/cases/#elifdefc2x_error.c"
int
2 errors generated.
```
And Gcc genete this:
```cpp
# 10 "test/cases/#elifdefc2x_error.c"
int
# 19 "test/cases/#elifdefc2x_error.c"
int
```
I think the behaviour of gcc looks more correct, i'm going to commit to the pull request to reflect this.

Comment (danielsan901998) [reply src/Preprocessor.zig]: If there are not more issues, I think that with commit e1567309b0af3e27dad70a22e083aecedc5c3183 it should be ready to merge.

Comment (Vexu) [reply src/Preprocessor.zig]: It looks like they have the same output with clang missing a newline after the `int`.

**Pull request merged**